ouroboros-network-framework/src/Ouroboros/Network/ConnectionManager/Core.hs

```diff
@@ -1263,6 +1266,7 @@ withConnectionManager ConnectionManagerArguments {
                             -- governor.
                     case outboundGovernorInfoChannel of
                       InResponderMode (Just infoChannel) | notifyOutboundGov
+                                                         , PeerSharingEnabled <- ownPeerSharing
```

Comment:
I'd move this to `notifyOutboundGov` just above, so all conditions are in one place.